nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -65,10 +65,13 @@ let
   authKeysFiles = let
     mkAuthKeyFile = u: nameValuePair "ssh/authorized_keys.d/${u.name}" {
       mode = "0444";
-      source = pkgs.writeText "${u.name}-authorized_keys" ''
-        ${concatStringsSep "\n" u.openssh.authorizedKeys.keys}
-        ${concatMapStrings (f: readFile f + "\n") u.openssh.authorizedKeys.keyFiles}
-      '';
+      source =
+        let inherit (u.openssh.authorizedKeys) keys keyFiles; in
+        assert !(lib.any (lib.hasInfix "\n") keys);
```

Comment:
this is better done with a different type on the `openssh.authorizedKeys` option, or if that's not feasible with a module-level assertion that also prints *which* key is invalid. while an assert does protect you from malformed keys it'll also remove all information about which key was faulty and where it was defined. (types preserve both, assertions can preserve the former easily.)

Reply:
Excellent idea! Let me experiment with it